client/angular-dapp/src/app/login/login.component.ts

```diff
@@ -39,17 +40,23 @@ export class LoginComponent implements OnInit {
     this.errored = false;
     this.unauthorized = false;
 
+    this.isAuthorized$.pipe(
+      filter(Boolean),
+      tap(() => this.isLoading = true),
+      switchMap(() => this.rolesService.get().pipe(finalize(() => this.isLoading = false)))
+    ).subscribe(roles => this.roles = roles);
+
     this.loginService.login(providerType).pipe(
       filter(({did}) => Boolean(did)),
-      switchMap(() => this.rolesService.get().pipe(finalize(() => this.isLoading = false))),
-    ).subscribe(roles => this.roles = roles, error => {
-      console.log(error);
```

Comment:
@dawidgil do you think that having the console.log here is/was useful? Or not really?

Reply:
Gosh, it was removed 😅